new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
-{{< /code >}}
+### Built-in Classes
 
-This rule will only be evaluated when the actor is a `User`; the `actor`
-argument is _specialized_ on that type. We could also use `matches` to express
-the same logic on an unspecialized rule:
+We said above that any class you use as a specializer
+(or on the right-hand side of the `matches` operator) must be
+registered with Polar. For convenience, Polar automatically
+registers a few _built-in_ classes, such as:
 
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor matches User{name: "alice"};
-{{< /code >}}
+* `Dictionary`
+* `String`
+* `List`
+* `Integer`
+* `Float`
 
-Either way, using the rule could look like this:
+These classes correspond to ones in the application language,
+e.g., in Python, `String` is actually the built-in `str` class.
 
-{{% exampleGet "specializedExample" %}}
+## Method Calls
 
-{{% callout "Note" "green" %}}
-Type specializers automatically respect the **inheritance** hierarchy of
-application classes. See the [Resources with
-Inheritance](guides/more/inheritance) guide for an in-depth
-example of how this works.
-{{% /callout %}}
+In addition to accessing the fields of application objects in a rule,
+you can also call methods on them. We'll demonstrate this using the
+built-in `String` class, but it applies to instances of any registered
+class.
 
-Once a class is registered, class or static methods can also be called from Oso
-policies:
+Suppose we wanted to match a range of actions that all started
+with the letter "r". (A silly encoding, perhaps, but illustrative.)
+We could allow all such actions using a rule like this:
 
-{{< code file="policy.polar" >}}
-allow(actor: User, action, resource) if actor.name in User.superusers();
-{{< /code >}}
+```polar
+allow(actor, action: String, resource) if
+    action.startswith("r");
+```
 
-{{% exampleGet "classMethodExample" %}}
+This calls the {{< lang >}} method `str.startswith`, guarding
+against invalid calls with the `String` specializer.
 
-### Built-in Types
+## Other Rules
 
-Methods called on the Polar built-in types `String`, `Dictionary`, `Number`,
-and `List` punt to methods on the corresponding application language class.
-That way you can use familiar methods like `.{{% exampleGet "startswith" %}}()`
-on strings regardless of whether they originated in your application or as a
-literal in your policy. This applies to all of Polar's [supported
-types](polar-syntax#primitive-types) in any supported application language. For
-examples using built-in types, see [the {{% exampleGet "langName" %}}
-library](reference/polar/classes) guide.
+So far all of our examples have involved only `allow` rules
+with three parameters. That's a natural place to start with Oso,
+because the `is_allowed` function generates queries of the form
+`allow(actor, action, resource)`. But you can write rules for
+whatever you like, and use them from within your `allow` rules.
```

Comment:
I think it's worth adding one sentence somewhere in this section that lets the reader know they can also use the `query_rule()` method to query arbitrary rules directly from the application. Perhaps at the very end.